Automattic/wp-calypso
# Add Jetpack bi-yearly products to `calypso-products` package.

This PR adds Jetpack Security T1 Bi-Yearly and VaultPress Backup T1 Bi-Yearly products/plans to the `calypso-products` package.



Asana Task: 1201210512993648-as-1204659654942256/f

## Proposed Changes

* Added `PRODUCT_JETPACK_BACKUP_T1_BI_YEARLY` to `products-list.ts`.
* Added `PLAN_JETPACK_SECURITY_T1_BI_YEARLY` to `plans-list.ts`.
* Added above 2 constants ⬆️ to `jetpack.ts` file.
* Updated `calypso-products' package unit tests, so that all passing.

## Testing Instructions

- First, sandbox `public-api.wordpress.com`.
- In your sandbox, open file: `wp-content/admin-plugins/wpcom-billing/default-shopping-cart-json-serializer.php`.
- Comment out lines 1050 - 1052 and save file, so that we can view Jetpack **Bi-Yearly** products at checkout.  It should looks like this:
```
// if ( 'jetpack' === $product->product_type && $product->renews_biyearly() ) {
// 	return null;
// }
```
- Go to https://wordpress.com/checkout/jetpack/jetpack_backup_t1_bi_yearly and https://wordpress.com/checkout/jetpack/jetpack_security_t1_bi_yearly
- Notice that the billing term variation dropdown is **Not** showing and the "**Included with your purchase**" features are **Not** showing.  This PR fixes this..  (See "Before" screenshot below.)
- Now checkout this branch/PR and spin up Calypso (`yarn start`).
- Now reload the checkout pages listed above and verify the billing term variation dropdown **_is_** showing and the "**Included with your purchase**" features **are now showing**. (See "After" screenshot below.)
- Run: `yarn test-packages calypso-products` and verify all tests are passing.

### Screenshots

Screenshot example for Jetpack Security (10GB) Bi-yearly (slug: `jetpack_security_t1_bi_yearly`)

BEFORE | AFTER
--- | ---
![Markup 2023-05-24 at 15 05 29](https://github.com/Automattic/wp-calypso/assets/11078128/64136211-8810-4572-8789-d39c09a31f9c) | ![Markup 2023-05-24 at 15 07 37](https://github.com/Automattic/wp-calypso/assets/11078128/9f403440-907d-432a-befe-1a7e90e4c9ba)


## Pre-merge Checklist
